sdk/src/main/java/io/opentelemetry/sdk/metrics/AbstractBoundInstrument.java

```diff
@@ -16,32 +16,57 @@
 
 package io.opentelemetry.sdk.metrics;
 
-import io.opentelemetry.metrics.LabelSet;
+import java.util.concurrent.atomic.AtomicLong;
 
-class AbstractBoundInstrument {
-  private final LabelSet labels;
+abstract class AbstractBoundInstrument {
+  // Atomically counts the number of references (usages) while also keeping a state of
+  // mapped/unmapped into a registry map.
+  private final AtomicLong refCountMapped;
+  private final Aggregator<?> aggregator;
 
-  AbstractBoundInstrument(LabelSet labels) {
-    this.labels = labels;
-    // todo: associate with an aggregator/accumulator
+  AbstractBoundInstrument(Aggregator<?> aggregator) {
+    this.aggregator = aggregator;
+    this.refCountMapped = new AtomicLong(0);
   }
 
-  @Override
-  public boolean equals(Object o) {
-    if (this == o) {
-      return true;
-    }
-    if (!(o instanceof AbstractBoundInstrument)) {
+  /**
+   * Returns {@code true} if the entry is still mapped and increases the reference usages, if
+   * unmapped returns {@code false}.
+   *
+   * @return {@code true} if successful.
+   */
+  boolean ref() {
```

Comment:
how about "bind" and "release" as names, rather than ref/unref? Cryptically short method names make me sad.

Reply:
you Java person :))

Reply:
done.

Reply:
done.